main/src/main/python/embedding_generator/main.py

```diff
@@ -0,0 +1,119 @@
+from transformers import DebertaModel, AutoTokenizer
+import torch
+import numpy as np
+from datasets import load_dataset
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--model_type', dest='model_type', type=str)
+parser.add_argument('--tokenizer', dest='tokenizer', type=str)
+parser.add_argument('--context_size', dest='context_size', type=int)
+parser.add_argument('--device', dest='device', type=str)
+parser.add_argument('--huggingface_dataset', dest='huggingface_dataset', type=str)
+parser.add_argument('--huggingface_subpart', dest='huggingface_subpart', type=str)
+parser.add_argument('--number_of_tokens', dest='number_of_tokens', type=int)
+parser.add_argument('--output', dest='output', type=str)
+args = parser.parse_args()
+
+context = int(args.context_size) # 512
+model_type = args.model_type # microsoft/deberta-base
+tokenizer = args.tokenizer # microsoft/deberta-base
+device = args.device # cuda
+huggingface_dataset = args.huggingface_dataset # wikipedia
+huggingface_subpart = args.huggingface_subpart # wikipedia
+number_of_tokens = int(args.number_of_tokens) # 10000000
+output = str(args.output) # deberta_embd
+
+torch.set_default_device(device)
+
+model = DebertaModel.from_pretrained(model_type).to(device)
+tokenizer = AutoTokenizer.from_pretrained(tokenizer)
+
+if huggingface_dataset:
+    if huggingface_subpart:
+        dataset = load_dataset(huggingface_dataset, huggingface_subpart).with_format(type="torch")
+    else:
+        dataset = load_dataset(huggingface_dataset).with_format(type="torch")
+else:
+    dataset = load_dataset('json', data_files={'train': 'text_data.json'})
+
+tokens_so_far = 0
+crr = 0
+
+sum_tensors = {}
+crr_tensors = {}
+
+i = 0
+
+print(len(dataset))
+
+while tokens_so_far < number_of_tokens:  # 10 million
+
+    index = np.random.randint(0, len(dataset))
+
+    text = dataset['train'][index]['text']
+
+    tokenized = tokenizer(text, return_tensors='pt')
+
+    tokens = tokenized['input_ids'][0]
+
+    if len(tokens) <= context + 1:
+
+        crr += 1
+        tokens_so_far += len(tokens)
+
+        sliced_output = {key: value[:, :] for key, value in tokenized.items()}
+
+    else:
+
+        i = np.random.randint(0, len(tokens) - context - 1)
+
+        crr += 1
+        tokens_so_far += context
```

Comment:
I don't understand what tokenizer() returns or how this subsampling of tokenized.tiems works, but I'll take your word for it.